tools/auth/kakao.go

```diff
@@ -0,0 +1,50 @@
+package auth
+
+import (
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/kakao"
+)
+
+var _ Provider = (*Kakao)(nil)
+
+// NameKakao is the unique name of the Kakao provider.
+const NameKakao string = "kakao"
+
+// Kakao allows authentication via Kakao OAuth2.
+type Kakao struct {
+	*baseProvider
+}
+
+// NewKakaoProvider creates a new Kakao provider instance with some defaults.
+func NewKakaoProvider() *Kakao {
+	return &Kakao{&baseProvider{
+		scopes:     []string{"profile_nickname", "profile_image", "account_email", "name", "gender", "age_range", "birthyear"},
```

Comment:
We need only the scopes that give us access to the mapped fields below (I guess account_email, name, and profile_image).

Reply:
That's right. the scope specs were different from my understanding.